lib/cloudregister/registerutils.py

```diff
@@ -503,6 +503,23 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def refresh_registry_credentials():
+    """Refresh registry credentials."""
+    requests.packages.urllib3.disable_warnings() # to silence InsecureRequestWarning
+    current_server = get_current_smt()
```

Comment:
`get_current_smt` calls `‎get_smt_from_store` which rewrites the local smt settings. IMO that's an unwanted side effect.

Reply:
I'm not sure it rewrites the local SMT settings ?
`get_smt_from_store` does create an SMT instance from the stored data, 
and `get_current_smt` makes sure that it matches /etc/hosts and that it is registered

is there another step I skipped on those methods ?

Reply:
do you mean that the stored data gets deleted if the stored data does not match `/etc/hosts` ?

Reply:
`get_smt_from_store()` loads the current target update server, it will not modify any data. That said, he code path activated does not guarantee that we find a replacement update server if the current server should not be reachable. Meaning as currently implemented we try the current target update server and if for some reason it is not reachable then we give up an tell the user that we failed. That is terrible. more in the next comment......